python/ray/rllib/ppo/ppo.py

```diff
@@ -228,52 +211,75 @@ def standardized(value):
             "shuffle_time": shuffle_time,
             "load_time": load_time,
             "sgd_time": sgd_time,
-            "sample_throughput": len(trajectory["observations"]) / sgd_time
+            "sample_throughput": len(samples["observations"]) / sgd_time
         }
 
-        print("kl div:", kl)
-        print("kl coeff:", self.kl_coeff)
-        print("rollouts time:", rollouts_time)
-        print("shuffle time:", shuffle_time)
-        print("load time:", load_time)
-        print("sgd time:", sgd_time)
-        print("sgd examples/s:", len(trajectory["observations"]) / sgd_time)
-        print("total time so far:", time.time() - self.start_time)
+        self.filter_manager.synchronize(self.remote_evaluators)
+        res = self._fetch_metrics_from_remote_evaluators()
+        res = res._replace(info=info)
+
+        return res
+
+    def _fetch_metrics_from_remote_evaluators(self):
+        episode_rewards = []
+        episode_lengths = []
+        metric_lists = [a.get_completed_rollout_metrics.remote()
+                        for a in self.remote_evaluators]
+        for metrics in metric_lists:
+            for episode in ray.get(metrics):
+                episode_lengths.append(episode.episode_length)
+                episode_rewards.append(episode.episode_reward)
+        avg_reward = (
+            np.mean(episode_rewards) if episode_rewards else float('nan'))
+        avg_length = (
+            np.mean(episode_lengths) if episode_lengths else float('nan'))
+        timesteps = np.sum(episode_lengths) if episode_lengths else 0
+
+        if self.file_writer:
+            traj_stats = tf.Summary(value=[
+                tf.Summary.Value(
+                    tag="ppo/rollouts/mean_reward",
+                    simple_value=avg_reward),
+                tf.Summary.Value(
+                    tag="ppo/rollouts/traj_len_mean",
+                    simple_value=avg_length)])
+            self.file_writer.add_summary(traj_stats, self.global_step)
+        self.global_step += 1
```

Comment:
Can we remove this (isn't it just self.iteration?)

Reply:
I had the same thought but global_step is actually incremented every STD iteration